src/calcurse.c

```diff
@@ -154,7 +154,7 @@ static inline void key_add_item(void)
 
 static inline void key_edit_item(void)
 {
-	if (wins_slctd() == APP && !event_dummy()) {
+	if (wins_slctd() == APP && !event_dummy(ui_day_get_sel())) {
```

Comment:
Huh? This looks like a bug fix to me that should not belong in this commit. It should be part of the commit that introduced the faulty code. In fact, I would prefer keeping `ui_day_dummy()` and using it already in the first commit that introduced `event_dummy()` here. This should allow us to get rid of all the stray bug fixes.

Reply:
Why I you reviewing commits that were long since replaced? This and your two previous comments concern outdated sources. Please take the commits of 17 March as the starting point.

Reply:
Most likely because I am not using GitHub review in the way I should. That's what GitHub gave me when clicking on the commit summary of this PR.

The very same issues are still there in the most recent version of the PR, aren't they?

Reply:
No!

Reply:
_In answer to this and the two previous comments_:

The function `ui_day_dummy()` was introduced from the start. Its purpose was to tell whether the selected item (in the appointments panel) is the dummy event; it had no arguments. Later it got a single one, as a need turned up for testing any item, not just the selected one. This argument is needed in my private version only (where development takes place), but can be left out of the GitHub version. Still later the function was renamed to `event_dummy(struct day_item *)` and moved from `ui-day.c` to `event.c` where the external variable `struct event dummy` is declared.

During the process of rebasing and squashing commits before upload, conflicts of course turned up that had to be fixed. When commits are reordered in interactive rebasing, function name changes are a particularly nasty business. I can no longer recall what happened or what I did, but `ui_day_dummy()` should have disappeared completely. In stead it survived in commit 7cba46d without being called and was removed in the following cf86e9e; `event_dummy()` was defined as `int event_dummy(struct day_item *item) {...}`, but called without argument; the forward declaration in `calcurse.h` was `int event_dummy();`. The code compiles, and I must have assumed that everything was OK. The same state remained in the following commit cf86e9e, before being corrected in 7c6488d.

I propose the following:

1. `ui_day_dummy()` is removed.
2. The `event_dummy()` function keeps its argument (it may come in handy), and the declaration in `calcurse.h` is corrected.
3. The faulty calls of `event_dummy()` are corrected.

Reply:
I agree with your analysis and your suggestion. Keeping the version with an argument seems to make sense, particularly when you say there are other patches needing this (that you may or may not want to merge later).